plugins/modules/docker_container.py

```diff
@@ -2403,6 +2419,19 @@ def has_different_configuration(self, image):
                         # expected_healthcheck comparison in this case.
                         continue
 
+                    if key == 'expected_labels' and self.parameters.image_label_mismatch == 'fail':
+                        # If there are labels from the base image that should be removed and
+                        # base_image_mismatch is fail we want raise an error.
+                        image_labels = self._get_image_labels(image)
+                        would_remove_labels = []
+                        for label in image_labels.keys():
```

Comment:
Iterating over a map iterates over the map keys automatically:
```suggestion
                        for label in image_labels:
```